src/generate.ts

```diff
@@ -149,26 +150,83 @@ function generatePostContent(
   return content;
 }
 
-function generatePrivate(
+async function generatePrivate(
   ssb: any,
   seed: string,
   i: number,
   latestmsg: number,
   msgsByType: MsgsByType,
+  tribesByAuthors: TribesByAuthor,
   author: Author,
   authors: Array<Author>,
-): string {
-  const content: Privatable<PostContent> = generatePostContent(
-    seed,
-    i,
-    latestmsg,
-    msgsByType,
-    authors,
-    'private',
-  );
-  const recps = generateRecipients(seed, author, authors);
-  content.recps = recps;
-  return ssbKeys.box(content, content.recps);
+): Promise<Msg | Privatable<Content>> {
+  const type = sampleCollection(seed, freq.PRIVATE_FREQUENCIES);
+  if (type === 'tribe_creation' || tribesByAuthors.size === 0) {
+    const {groupId, groupInitMsg} = await pify<any>(ssb.tribes.create)(null);
+    const groupIds = tribesByAuthors.get(author.id) ?? new Set();
+    groupIds.add(groupId);
+    tribesByAuthors.set(author.id, groupIds);
+    // console.log(
+    //   `${author.id.slice(0, 6)} created tribe ${groupId.slice(0, 6)}`,
+    // );
+    return groupInitMsg as Msg;
+  } else if (
+    type === 'tribe_invitation' &&
+    tribesByAuthors.has(author.id) &&
+    authors.length > 1
+  ) {
+    const groupId = uniformSample(
+      seed,
+      Array.from(tribesByAuthors.get(author.id)?.keys() ?? []),
+    );
+    let inviteeId: FeedId;
+    do {
+      inviteeId = paretoSample(seed, authors).id;
+    } while (inviteeId === author.id);
+    const text = __lorem.generateWords(randomInt(seed, 1, 9));
+    const msg = await pify(ssb.tribes.invite)(groupId, [inviteeId], {text});
+    const groupIds = tribesByAuthors.get(inviteeId) ?? new Set();
+    groupIds.add(groupId);
+    tribesByAuthors.set(inviteeId, groupIds);
+    // console.log(
+    //   `${author.id.slice(0, 6)} invited ${inviteeId.slice(
+    //     0,
+    //     6,
+    //   )} to tribe ${groupId.slice(0, 6)}`,
+    // );
+    return msg as Msg;
+  } else if (type === 'tribe_message' && tribesByAuthors.has(author.id)) {
+    const content: Privatable<PostContent> = generatePostContent(
+      seed,
+      i,
+      latestmsg,
+      msgsByType,
+      authors,
+      'private',
+    );
+    const groupId = uniformSample(
+      seed,
+      Array.from(tribesByAuthors.get(author.id)?.keys() ?? []),
+    );
+    const recps = [groupId, author.id];
+    content.recps = recps;
+    // console.log(
+    //   `${author.id.slice(0, 6)} posted to tribe ${groupId.slice(0, 6)}`,
+    // );
+    return content;
+  } else {
+    const content: Privatable<PostContent> = generatePostContent(
+      seed,
+      i,
+      latestmsg,
+      msgsByType,
+      authors,
+      'private',
+    );
+    const recps = generateRecipients(seed, author, authors);
```

Comment:
Havent reviewed this function. Don't know what authors is. Are they the other feeds?
Anyway I assume this generates an array of feedIds

Reply:
Correct and correct!